pkg/parser/parser.go

```diff
@@ -0,0 +1,121 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/tjgurwara99/monkey-go/pkg/ast"
+	"github.com/tjgurwara99/monkey-go/pkg/token"
+)
+
+type parser struct {
+	tokens chan token.Token
+
+	current token.Token
+	peek    token.Token
+
+	errors []error
+}
+
+func New(t chan token.Token) *parser {
+	return &parser{
+		tokens: t,
+		errors: []error{},
+	}
+}
+
+func (p *parser) Errors() []error {
+	return p.errors
+}
+
+func (p *parser) nextToken() {
+	p.current = p.peek
+	p.peek = <-p.tokens
+}
+
+func (p *parser) ParseProgram() *ast.Program {
+	program := &ast.Program{
+		Statements: []ast.Statement{},
+	}
```

Comment:
Should we not have a constructor to return this - the current implementation requires us to know the details of `ast.Program` struct.